fenix/app/src/main/java/org/mozilla/fenix/extension/WebExtensionPromptFeature.kt

```diff
@@ -79,13 +106,64 @@ class WebExtensionPromptFeature(
 
     private fun handlePermissionRequest(
         addon: Addon,
-        promptRequest: WebExtensionPromptRequest.Permissions,
+        promptRequest: WebExtensionPromptRequest.AfterInstallation.Permissions,
+    ) {
+        if (hasExistingPermissionDialogFragment()) return
+        showPermissionDialog(
+            addon,
+            promptRequest,
+        )
+    }
+
+    @VisibleForTesting
+    internal fun handleInstallationFailedRequest(
+        exception: WebExtensionInstallException,
     ) {
-        if (hasExistingPermissionDialogFragment()) {
-            return
+        val addonName = exception.extensionName ?: ""
+        var title = context.getString(R.string.mozac_feature_addons_failed_to_install, "")
+        val message = when (exception) {
+            is WebExtensionInstallException.Blocklisted -> {
+                context.getString(R.string.mozac_feature_addons_blocklisted, addonName)
+            }
+
+            is WebExtensionInstallException.Unknown, is WebExtensionInstallException.UserCancelled -> {
```

Comment:
Why is `UserCancelled` getting an error message? I thought we didn't want any message when that happens?

Reply:
It's simulating the same behaviour that we had when we had the snackbar. I guess we should remove it?

Reply:
Removed!